### ISSUE TITLE
403 on Grafana v8.3.6 on main branch

### ISSUE THREAD

**Issue by midN** [lightstep/lightstep-grafana-plugin#97]
Getting 403 on Main branch and latest release: 1.2.0.
```
method=GET path=/api/datasources/proxy/3/projects/XXXX/streams status=403
```

The API Key is definitely valid, works directly on API and on monkey-patched branch https://github.com/lightstep/lightstep-grafana-plugin/tree/enceeobee-fix-annotation-query

P.S The older plugins produces
```
method=GET path=/api/datasources/proxy/3/public/v0.2/YYYY/projects/XXXX/streams
```

UPDATE: In particular, this broke it https://github.com/lightstep/lightstep-grafana-plugin/commit/395726ab6352b086df701843c29e240cb6ab732b
The API Key is removed, URL changed and the API key is never re-used called. cc @zeitlinger @miry

**Comment by zeitlinger** [comment]
did you put the API key in [secureJsonData](https://github.com/lightstep/lightstep-grafana-plugin/blob/395726ab6352b086df701843c29e240cb6ab732b/src/lightstep-datasource/plugin.json#L44)?

**Comment by smithclay** [comment]
see example here for setting `secureJsonData`: https://github.com/lightstep/lightstep-grafana-plugin/blob/master/provisioning/datasources/lightstep.yaml#L20-L21

**Comment by alongfield** [comment]
@zeitlinger This broke existing manually created datasources as well for Grafana 8.3.5.  The last working version is 1.1.19 for me.

**Comment by midN** [comment]
@zeitlinger @smithclay There are cases where yaml/ini is not accessible or easily modifiable, such as running from k8s.

Is there a reason this cannot be modified in UI like other plugins?
Having a plugin that needs to be configured in config is quite clumbersome.

**Comment by zeitlinger** [comment]
@midN I've found an [example](https://github.com/pass-culture/pc-monitoring/blob/a06558db6b6a52ca69242caa505f22e530bebaa3/public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx) where the credentials are updated from the UI - so this should work.

The example uses react (which is the new style for grafana plugins) which I'm not familiar with.

Maybe @smithclay can help. PR is very welcome as well :smile:

**Comment by zeitlinger** [comment]
> @zeitlinger This broke existing manually created datasources as well for Grafana 8.3.5. The last working version is 1.1.19 for me.

Does it work when you add the API key to [secureJsonData](https://github.com/lightstep/lightstep-grafana-plugin/blob/master/provisioning/datasources/lightstep.yaml#L20-L21)?

**Comment by zeitlinger** [comment]
@smithclay can you (as maintainer) help?

**Comment by smithclay** [comment]
We're triaging this and will have an update out shortly.

For anyone impacted by this, in the meantime use [v1.1.9](https://github.com/lightstep/lightstep-grafana-plugin/releases/tag/v1.1.9). The v1.2.x release and current master branch is impacted.

**Comment by smithclay** [comment]
Now released: https://github.com/lightstep/lightstep-grafana-plugin/releases/tag/v1.2.2

Note that because we changed the API key to use a secure field in v1.2.0, if you are upgrading you will need to set the API key again (either in the UI or using a provisioning file). Thanks all for the detailed bug reports and context -- this should work for anyone who is using provisioning files _or_ the UI to set the API key. The "Save and Test" button on the configuration page is now fixed as well.
